### PR TITLE
Update vsxmake and fixes for IntelliSense

### PR DESCRIPTION
**Pull request by WubiCookie**

### CODE REVIEW

File: xmake/plugins/project/vsxmake/getinfo.lua [hunk 1]
```diff
@@ -158,12 +158,24 @@ function _make_targetinfo(mode, arch, target)
...
+    -- defaults to c++14
+    targetinfo.cppstd        = "stdcpp14"
+    for _,language in ipairs(languages) do
+        local lowerlanguage = language:lower()
+        if lowerlanguage:find("c++") or lowerlanguage:find("cxx") or lowerlanguage:find("cpp") then
+            -- fix c++ and cxx to cpp for #target#.vcxproj
+            targetinfo.cppstd = "std" .. lowerlanguage:replace("c++", "cpp", {plain = true}):replace("cxx", "cpp", {plain = true})
+            break
+        end
+    end
```
Comment (waruqi) [inline xmake/plugins/project/vsxmake/getinfo.lua:177]: we need not it too.

File: xmake/plugins/project/vsxmake/vsproj/templates/vcxproj/XmakeConfig(mode,arch) [hunk 2]
```diff
@@ -1,3 +1,11 @@
+  <PropertyGroup Label="Configuration" Condition="'$(Configuration)|$(Platform)'=='#mode#|#vsarch#'">
+    <PlatformToolset>#toolset_version#</PlatformToolset>
+  </PropertyGroup>
+  <ItemDefinitionGroup Condition="'$(Configuration)|$(Platform)'=='#mode#|#vsarch#'">
+    <ClCompile>
+      <LanguageStandard>#cppstd#</LanguageStandard>
+    </ClCompile>
+  </ItemDefinitionGroup>
```
[the remark this answers is not in the document]
Comment (waruqi) [reply xmake/plugins/project/vsxmake/vsproj/templates/vcxproj/XmakeConfig(mode,arch):8]: https://github.com/xmake-io/xmake/blob/028232bde5a7f829c7865b94322275ba6eff0642/xmake/plugins/project/vsxmake/vsproj/templates/vcxproj/XmakeConfig(mode%2Carch)#L10

File: xmake/plugins/project/vsxmake/vsproj/templates/vcxproj/XmakeConfig(mode,arch) [hunk 3]
```diff
@@ -1,3 +1,6 @@
+  <PropertyGroup Label="Configuration" Condition="'$(Configuration)|$(Platform)'=='#mode#|#vsarch#'">
+    <PlatformToolset>#toolset_version#</PlatformToolset>
+  </PropertyGroup>
```
Comment (waruqi) [inline xmake/plugins/project/vsxmake/vsproj/templates/vcxproj/XmakeConfig(mode,arch):3]: we need not it too. Xmake.props has added it.

https://github.com/xmake-io/xmake/blob/a70d5148d402195a411c590fd049c045ed5b5c7b/xmake/plugins/project/vsxmake/vsproj/Xmake.props#L54-L61

Comment (waruqi) [reply xmake/plugins/project/vsxmake/vsproj/templates/vcxproj/XmakeConfig(mode,arch):3]: `<PlatformToolset Condition="'$(VisualStudioVersion)' == '17.0'">v143</PlatformToolset>` has been added on master/dev. 

see https://github.com/xmake-io/xmake/issues/1838

File: xmake/plugins/project/vsxmake/vsproj/templates/vcxproj/XmakeConfig(mode,arch) (hunk 2 above)
[the remark this answers is not in the document]
Comment (WubiCookie) [reply xmake/plugins/project/vsxmake/vsproj/templates/vcxproj/XmakeConfig(mode,arch)]: I found why I needed to add this. I do `set_languages("c++latest")` in my xmake.lua, but the props look for `cxx20`...
btw, `stdcpp20` is a valid value for LanguageStandard now.
( @SirLynix that's why I told you IntelliSense was broken for me )

[the remark this answers is not in the document]
Comment (WubiCookie) [reply xmake/plugins/project/vsxmake/vsproj/templates/vcxproj/XmakeConfig(mode,arch)]: okay, I have to use `set_languages("cxx20", "c++latest")` in order to have everything working fine... (std::format requires cpplatest, providing only cxx20 is not enough)

**Pull request merged**